target/linux/ramips/dts/mt7628an_asus_rt-ac1200.dts

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+// Copyright (c) 2022 Ray Wang
+// Copyright (c) 2022 Ivan Pavlov
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rt-ac1200", "mediatek,mt7628an-soc";
+	model = "Asus RT-AC1200";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "blue:power";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "blue:wlan2g";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		usb {
+			label = "blue:usb";
+			gpios = <&gpio 6 GPIO_ACTIVE_HIGH>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		led_all {
```

Comment:
Ditto.

Reply:
Ditto

Reply:
Hi, is there a naming convention for property `gpio-export,name` under this node? The naming scheme in the current device trees seems all over the place; some has underscores, some has dashes, ans some has none:
```
PS C:\OpenWRT\github\openwrt\target\linux\ramips\dts> findstr "gpio-export,name" *.*
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "modem-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "modem-rf-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "rtc-enable";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim-select";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim1-detect";
mt7620a_alfa-network_r36m-e4g.dts:                      gpio-export,name = "sim2-detect";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "modem-enable";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "modem-rf-enable";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim-select";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim1-detect";
mt7620a_alfa-network_tube-e4g.dts:                      gpio-export,name = "sim2-detect";
mt7620a_dlink_dwr-118-a1.dts:                   gpio-export,name = "usb";
mt7620a_dlink_dwr-118-a2.dts:                   gpio-export,name = "usb";
mt7620a_dovado_tiny-ac.dts:                     gpio-export,name = "usbpower";
mt7620a_edimax_br-6478ac-v2.dts:                        gpio-export,name="usb-power";
mt7620a_hiwifi_hc5761.dts:                      gpio-export,name = "usbpower";
mt7620a_hiwifi_hc5861.dts:                      gpio-export,name = "usbpower";
mt7620a_hiwifi_hc5861.dts:                      gpio-export,name = "sdpower";
mt7620a_hnet_c108.dts:                  gpio-export,name = "power_modem";
mt7620a_lava_lr-25g001.dts:                     gpio-export,name = "usbpower";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb0";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb1";
mt7620a_lenovo_newifi-y1s.dts:                  gpio-export,name = "usb2";
mt7620a_sanlinking_d240.dts:                    gpio-export,name = "power_mpcie2";
mt7620a_sanlinking_d240.dts:                    gpio-export,name = "power_mpcie1";
mt7620a_sercomm_na930.dts:                      gpio-export,name = "telit";
mt7620a_sitecom_wlr-4100-v1-002.dts:                    gpio-export,name = "usb-power";
mt7620a_tplink_archer-mr200.dts:                        gpio-export,name = "power_usb1";
mt7620a_yukai_bocco.dts:                        gpio-export,name = "rec";
mt7620a_yukai_bocco.dts:                        gpio-export,name = "play";
mt7620a_zyxel_keenetic-viva.dts:                        gpio-export,name = "usb";
mt7620n_dlink_dwr-921-c1.dts:                   gpio-export,name = "lte_modem_enable";
mt7620n_dlink_dwr-922-e2.dts:                   gpio-export,name = "lte_modem_enable";
mt7620n_zyxel_keenetic-omni-ii.dts:                     gpio-export,name = "usb";
mt7620n_zyxel_keenetic-omni.dts:                        gpio-export,name = "usb";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "m2-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie0-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie0-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie1-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie1-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie2-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "minipcie2-reset";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "pcie-perst-enable";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "sim-select";
mt7621_alfa-network_quad-e4g.dts:                       gpio-export,name = "watchdog-enable";
mt7621_d-team_newifi-d2.dts:                    gpio-export,name = "power_usb3";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_usb2";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_usb3";
mt7621_d-team_pbr-m1.dts:                       gpio-export,name = "power_sata";
mt7621_humax_e10.dts:                   gpio-export,name = "power:usb";
mt7621_lenovo_newifi-d1.dts:                    gpio-export,name = "usb2power";
mt7621_lenovo_newifi-d1.dts:                    gpio-export,name = "usb3power";
mt7621_mikrotik_routerboard-7xx.dtsi:                   gpio-export,name = "buzzer";
mt7621_mikrotik_routerboard-7xx.dtsi:                   gpio-export,name = "usb_power";
mt7621_sercomm_na502.dts:                       gpio-export,name = "zwave_reset";
mt7621_sercomm_na502.dts:                       gpio-export,name = "zigbee_reset";
mt7621_tplink_eap235-wall-v1.dts:                       gpio-export,name = "poe-passthrough";
mt7621_unielec_u7621-01.dtsi:                   gpio-export,name = "modem_reset";
mt7621_unielec_u7621-06.dtsi:                   gpio-export,name = "modem_reset";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_usb2";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_usb3";
mt7621_xzwifi_creativebox-v1.dts:                       gpio-export,name = "power_sata";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "4g1-pwr";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "4g2-pwr";
mt7621_zbtlink_zbt-wg1602.dtsi:                 gpio-export,name = "ext-usb";
mt7628an_asus_rt-ac1200.dts:                    gpio-export,name = "led_all";
mt7628an_glinet_gl-mt300n-v2.dts:                       gpio-export,name = "usb";
mt7628an_hak5_wifi-pineapple-mk7.dts:                   gpio-export,name = "usb-power";
mt7628an_hiwifi_hc5761a.dts:                    gpio-export,name = "usb_power";
mt7628an_minew_g1-c.dts:                        gpio-export,name = "ws2812";
mt7628an_minew_g1-c.dts:                        gpio-export,name = "nrf_power";
mt7628an_zyxel_keenetic-extra-ii.dts:                   gpio-export,name = "usbpower";
rt3050_arcwireless_freestation5.dts:                    gpio-export,name = "poe-passthrough";
rt3052_accton_wr6202.dts:                       gpio-export,name = "usb";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_data";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_clock";
rt3052_jcg_jhr-n926r.dts:                       gpio-export,name = "display_blank";
rt3052_planex_mzk-wdpr.dts:                     gpio-export,name = "lcd_ctrl1";
rt3662_dlink_dir-645.dts:                       gpio-export,name = "usb";
rt3662_edimax_br-6475nd.dts:                    gpio-export,name="usb";
rt3662_engenius_esr600h.dts:                    gpio-export,name = "usb";
rt3662_omnima_hpm.dts:                  gpio-export,name = "usb0";
rt3662_omnima_hpm.dts:                  gpio-export,name = "usb1";
rt3883_sitecom_wlr-6000.dts:                    gpio-export,name = "usb";
rt5350_dlink_dir-320-b1.dts:                    gpio-export,name = "usb";
rt5350_dlink_dir-320-b1.dts:                    gpio-export,name = "root_hub";
rt5350_dlink_dwr-512-b.dts:                     gpio-export,name = "slic_int";
rt5350_dlink_dwr-512-b.dts:                     gpio-export,name = "modem3g_enable";
rt5350_hame_mpr-a1.dts:                 gpio-export,name = "usb";
rt5350_hame_mpr-a1.dts:                 gpio-export,name = "root_hub";
rt5350_hame_mpr-a2.dts:                 gpio-export,name = "usb";
rt5350_hame_mpr-a2.dts:                 gpio-export,name = "root_hub";
rt5350_hilink_hlk-rm04.dts:                     gpio-export,name = "hlk-rm04:gpio0";
rt5350_hilink_hlk-rm04.dts:                     gpio-export,name = "hlk-rm04:gpio1";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio0";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio1";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio17";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio18";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio19";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio20";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio21";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio22";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio23";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio24";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio25";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio26";
rt5350_nixcore_x1.dtsi:                 gpio-export,name = "gpio27";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "relay1";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "relay2";
rt5350_olimex_rt5350f-olinuxino-evb.dts:                        gpio-export,name = "button";
rt5350_poray_x5.dts:                    gpio-export,name = "usb-mode";
rt5350_tenda_3g150b.dts:                        gpio-export,name = "usb";
rt5350_trendnet_tew-714tru.dts:                 gpio-export,name = "repeater_switch";
rt5350_trendnet_tew-714tru.dts:                 gpio-export,name = "wisp_switch";
rt5350_unbranded_a5-v11.dts:                    gpio-export,name = "usb";
rt5350_unbranded_a5-v11.dts:                    gpio-export,name = "root_hub";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio0";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio7";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio8";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio9";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio12";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio13";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio14";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio17";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio18";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio19";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio20";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio21";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio22";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio23";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio24";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio25";
rt5350_vocore_vocore.dtsi:                      gpio-export,name = "gpio26";
```

Reply:
AFAIK there's no such thing. The convention I said above is from [1], but it is not that everyone is strictly following the convention (even the upstream kernel is mixed up), and it does not say anything on property value strings anyway. No one did not proposed and enforced a good convention (yet), that's all.
Personally, I prefer dash to underscore in all cases except where dash cannot be used (e.g. label). [2]

[1] https://elinux.org/Device_Tree_Linux#Linux_conventions
[2] https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3

Reply:
Got it, thanks. I'll change it to a dash for the sake of consistency in the next (and hopefully final) commit, and (hopefully) address the LED blinking issue along with it.